### ISSUE TITLE
[Feature Request] - Proxy support

### ISSUE THREAD

**Issue by slax-creator** [veeso/termscp#222]
## Description

Proxy support would be a valuable feature for termscp as it would allow users to route their SFTP/SCP traffic through proxy servers, enhancing security and enabling access in network-restricted environments. Adding this feature to termscp would make it more versatile and useful in various network environments.

Thank you for considering this feature request, and please let me know if you have any questions or need further clarification.

### Changes

- **Proxy Support**: Implement the ability to configure and use proxy servers for SFTP and SCP connections. This should include support for various types of proxies, such as SOCKS and HTTP proxies.

## Implementation

1. **Proxy Configuration**: Add a dedicated section in the client's configuration file where users can specify proxy settings. This should include options for proxy type (e.g., SOCKS5, HTTP), proxy server address, port, and proxy authentication details. Additionally, the client should automatically detect and respect existing environment variables such as http_proxy, https_proxy, and others for proxy configuration, making it convenient for users who have these variables set in their environment.

2. **Command-Line Options**: Allow users to set proxy-related options via command-line arguments when invoking the client. For example, users should be able to specify proxy information when running SFTP or SCP commands.

3. **Fallback Mechanism**: Implement a fallback mechanism to ensure that the client can work without a proxy if no proxy settings are specified. It should be easy to switch between using a proxy and direct connections.

4. **Documentation**: Provide comprehensive documentation that explains how to configure and use the new proxy support feature. This should include examples and best practices for different proxy server types.

**Comment by farzadmf** [comment]
Wow, really nice!!! Someone opens an issue, simply ignore it until the bot closes it!!!!

**Comment by farzadmf** [comment]
Yeah, sorry @veeso ; you're right; I was a bit tired trying different things and see them not working because of this feature. I apologize.

Unfortunately, at least for now, I just needed something to work with my setup, without me needing to invest on it (of course, it doesn't help that I'm not super efficient with Rust).

I was so happy to find this, and then I realized it doesn't work with my setup, hence the message. I apologize again, and I just did what I hate other people doing, sorry!
